From cf76fd4ff92945d73435544941b2ba06268e0546 Mon Sep 17 00:00:00 2001
From: Mauro Rossi <issor.oruam@gmail.com>
Date: Tue, 12 Sep 2017 00:17:10 +0200
Subject: [PATCH 24/28] fd_utils: fix DEBUG mode booting issue

Porting to oreo-x86 codebase of nougat-x86 commit
bca7b1f81f "fd_utils: fix DEBUG mode booting issue"

Original commit message:

Remove the "/android/" prefix before the white list checking.

Suggested by Chris Vandomelen <chris@sightworks.com>.

Conflicts:
	core/jni/fd_utils.cpp

Change-Id: I18bdd4d29dfecbf42afdede37d904d50f3f13460
---
 core/jni/fd_utils.cpp | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/core/jni/fd_utils.cpp b/core/jni/fd_utils.cpp
index c5904e0e9e5e..5716ff5f56fb 100644
--- a/core/jni/fd_utils.cpp
+++ b/core/jni/fd_utils.cpp
@@ -213,10 +213,9 @@ FileDescriptorInfo* FileDescriptorInfo::CreateFromFd(int fd, std::string* error_
   std::string file_path;
   const std::string fd_path = android::base::StringPrintf("/proc/self/fd/%d", fd);
   if (!android::base::Readlink(fd_path, &file_path)) {
-    *error_msg = android::base::StringPrintf("Could not read fd link %s: %s",
-                                             fd_path.c_str(),
-                                             strerror(errno));
-    return nullptr;
+    return NULL;
+  } else if (!strncmp(file_path.c_str(), "/android/", 9)) {
+    file_path = file_path.substr(8);
   }
 
   if (!whitelist->IsAllowed(file_path)) {
-- 
2.17.1

